ros/src/sensing/drivers/camera/packages/vectacam/nodes/vectacam/VectaCam.cpp

```diff
@@ -0,0 +1,261 @@
+#include "VectaCam.h"
+
+VectaCam::VectaCam(unsigned int in_configuration_port, unsigned int in_data_port, std::string in_parameter_file)
+{
+	this->data_port_ 			= in_data_port;
+	this->configuration_port_ 	= in_configuration_port;
+	this->camera_ready_ 		= false;
+	this->running_ 				= false;
+	this->fps_ 					= 0.0;
+	this->frame_number_			= 0;
+	this->frame_counter_		= 0;
+	this->payload_offset_ 		= 16;
+	this->image_width_ 			= VECTACAM_IMG_WIDTH * VECTACAM_NUM_CAMERAS;
+	this->image_height_ 		= VECTACAM_IMG_HEIGHT;
+	this->image_buffer_ 		= new char[image_width_ * 3 * image_height_];
+	this->parameter_file_ 		= in_parameter_file;
+	_initialize_camera(configuration_port_, data_port_, parameter_file_);
+}
+
+void VectaCam::_parse_parameter_file(std::string in_parameter_file, std::vector<VectaCamCommand>& out_commands)
+{
+	std::string line;
+	std::ifstream config_file (parameter_file_);
+	std::string delimiter = ":";
+	out_commands.clear();
+	if (config_file.is_open())
+	{
+		while ( getline (config_file,line) )
+		{
+			std::vector<std::string> tokens;
+			boost::split(tokens, line, boost::is_any_of(":"));
+			if (tokens.size()==2)
+			{
+				VectaCamCommand current_command;
+				int full_address = std::stoi(tokens[0], NULL, 16);
+				current_command.low_nibble_address = (unsigned char)( (full_address>>0) & 0xFF);
+				current_command.high_nibble_address = (unsigned char)((full_address>>8) & 0xFF);
+				current_command.command_data = std::stoi(tokens[1], NULL, 16);
+				out_commands.push_back(current_command);
+			}
+			else
+				std::cout << "Unrecognized command in configuration file: " << parameter_file_ << std::endl;
+		}
+		config_file.close();
+	}
+	else
+		std::cout << "Unable to open file:" << parameter_file_;
+}
+
+void VectaCam::_send_commands_to_camera(unsigned int in_port, std::vector<VectaCamCommand> in_commands)
+{
+
+	int socket_descriptor;
+	struct sockaddr_in socket_address;
+	if ((socket_descriptor=socket(AF_INET, SOCK_DGRAM, IPPROTO_UDP)) < 0)
+	{
+		std::cout << "Problem creating socket\n";
+		return;
+	}
+	memset((char *) &socket_address, 0, sizeof(socket_address));
+	socket_address.sin_family = AF_INET;
+	socket_address.sin_port = htons(in_port);
+	if (inet_aton("10.0.0.1", &socket_address.sin_addr)==0)
+	{
+		std::cout << "Invalid IP address" << std::endl;
+		return;
+	}
+	for (unsigned int i=0; i< in_commands.size(); i++)
+	{
+		VectaCamCommand current_command = in_commands[i];
+		unsigned char command_buffer[12] = {0x05, 0x00, 0x36, 0x00, 0x00, 0x00,
+									current_command.low_nibble_address, current_command.high_nibble_address, current_command.command_data,
+									0x00, 0x00, 0x00};
+		unsigned int sent_data = sendto(socket_descriptor, command_buffer, sizeof(command_buffer), 0,(struct sockaddr *) &socket_address, sizeof(socket_address));
```

Comment:
Return value type of `sendto` is `ssize_t`. Please do not use `unsigned` type. Please see `man 2 sendto`.